tests/javascript/treeselectmenu/spec.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @package     Joomla.Tests
+ * @subpackage  JavaScript Tests
+ *
+ * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ *
+ * @since       3.6.3
+ * @version     1.0.0
+ */
+
+define(['jquery', 'testsRoot/treeselectmenu/spec-setup', 'jasmineJquery'], function ($) {
+	
+	describe('Treeselectmenu Expand/Collapse of a node', function () {
+
+		var $toggle = $('span.treeselect-toggle');
+
+		beforeEach(function () {
+			spyOnEvent('span.treeselect-toggle', 'click');
+			$('span.treeselect-toggle').click();
+		});
+		it('Should have trigger treeselect-toggle click', function () {
+			expect('click').toHaveBeenTriggeredOn('span.treeselect-toggle');
+		});
+		it('Should have parent node element to "ul.treeselect-sub"', function () {
+			expect($toggle.parent().find('ul.treeselect-sub')).toExist();
+		});
+		it('Should parent node element "ul.treeselect-sub" visible', function () {
+			expect($toggle.parent().find('ul.treeselect-sub').is(':visible')).toBe(true);
+		});
+	});
+
+	describe('Treeselectmenu Checks all checkboxes the tree', function () {
+		var spyEvent;
+
+		beforeEach(function () {
+			spyEvent = spyOnEvent('#treeCheckAll', 'click');
+			$('#treeCheckAll').click();
+		});
+
+		it('Should have trigger treeCheckAll click', function () {
+			expect('click').toHaveBeenTriggeredOn('#treeCheckAll');
```

Comment:
These test cases which (triggers a click and verify the click triggered) doesn't add a value for test coverage, it is important that, to check the expected outcome (e.g changing styles, creating an element, opening a window & etc.) tested after the trigger.